src/connect.ts

```diff
@@ -17,6 +17,8 @@ import { StateShareClient, StateShareServer } from './ipc-rpc-state-share';
 import ConnectionString, { CommaAndColonSeparatedRecord } from 'mongodb-connection-string-url';
 import EventEmitter from 'events';
 
+const isAtlas = (str: string) => str.match(/mongodb.net[:/]/i);
```

Comment:
Let’s maybe make this a bit more precise so that we don’t match `mongodb://supermongo.net/` or `mongodb://localhost/mongodb.net:foo`?

```suggestion
function isAtlas(str: string): boolean {
  try {
    const { hosts } = new ConnectionString(str);
    return hosts.every(host => /(^|\.)mongodb.net(:|$)/.test(host));
  } catch { return false; }
}
```